docs/topics/lombok.md

```diff
@@ -0,0 +1,149 @@
+[//]: # (title: Lombok compiler plugin)
+
+> The Lombok compiler plugin is [Experimental](components-stability.md). 
+> It may be dropped or changed at any time. Use it only for evaluation purposes. 
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-7112).
+>
+{type="warning"}
+
+Since 1.5.20, Kotlin supports the experimental Lombok Kotlin compiler plugin. This plugin allows the use of
+Java's Lombok-generated declarations in mixed Kotlin/Java sources.
+If Java's Lombok-generated declarations are in another module than the Kotlin code that uses these declarations, 
+then you don't need to use this plugin.
+
+[Lombok](https://projectlombok.org/) does Java code generation itself, not the Lombok Kotlin compiler plugin. 
+So, you still need to configure Lombok as usual, when using this plugin. 
+Learn more about [how to make the plugin seeing Lombok's config](#defining-the-place-of-lombok-config).
+
+## Support of annotations
+
+Learn which annotations the plugin supports and the
+[current state of Lombok Kotlin compiler plugin](https://github.com/JetBrains/kotlin/blob/master/plugins/lombok/lombok-compiler-plugin/README.md).
+
+> Generation of `@Builder` annotation won't be supported in the current prototype. However, you can give your feedback
+> about the plugin and vote for [`@Builder` in YouTrack](https://youtrack.jetbrains.com/issue/KT-46959).
+>
+{type="warning"}
+
+> Kotlin compiler ignores Lombok annotations if you use them in Kotlin code.
+>
+{type="note"}
+
+Please read below about how to apply the Lombok Kotlin compiler plugin to your Gradle/Maven build.
+
+## Gradle
+
+### Main settings
+
+Apply the `kotlin-plugin-lombok` Gradle plugin:
+
+```groovy
+plugins {
+    id "org.jetbrains.kotlin.plugin.lombok" version "%kotlinVersion%"
+}
+```
+
+Alternatively, you can use the `apply plugin` syntax:
+
+```groovy
+apply plugin: 'kotlin-plugin-lombok'
+```
+
+The Lombok itself should be set up as the Java annotation processor (not via kapt) or as the plugin as below:
+
+```groovy
+plugins {
+    id "io.freefair.lombok" version "5.3.0"
+}
+```
+Learn the [test project with examples of Lombok compiler plugin usage](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/nokapt).
+
+### Defining the place of lombok.config
+
+If you use the file `lombok.config` for [configuring Lombok](https://projectlombok.org/features/configuration),
+Kotlin Lombok compiler plugin should know where this file situates. Use the following syntax in your `build.gradle` file:
```

Comment:
"Kotlin Lombok compiler plugin should know" lacks actor. Please give a direct instruction: what the user should do.
smth like "provide a path to it to the plugin". 
Don't forget to add `(.kts)` after `build.gradle`

Reply:
>"Kotlin Lombok compiler plugin should know" lacks actor. Please give a direct instruction: what the user should do.

I'm trying to explain the underlying motivation for actions. Does it work?

>If you use the file `lombok.config` for [configuring Lombok](https://projectlombok.org/features/configuration),
Kotlin Lombok compiler plugin should know where this file situates. Provide a `lombok.config`'s path to the plugin in your `build.gradle` file:


>Don't forget to add (.kts) after build.gradle

But this is the Groovy file. I researched our compiler plugin's docs, in many (or even most of them) instructions in Groovy are used.

Reply:
> I'm trying to explain the underlying motivation for actions. Does it work?

I think it's unnecessary here. "If you use a config file, provide a path to it" is enough.

> But this is the Groovy file. 

It can be Kotlin, can't it? I can't remember cases where we restrict the usage of Kotlin for Gradle build scripts.
Compiler plugin docs were written some years ago when Kotlin support in build.gradle was poor and maybe even unofficial. We should update these docs as well.

Reply:
Ok, I'll fix to .kts in lombok.md.

>We should update these docs as well.

Ok, but until that time I'll use the example in Groovy in kapt.md for consistency.